sequence_task_2/student/include/impl.h

```diff
@@ -14,4 +14,7 @@
  * @note See https://en.cppreference.com/w/cpp/algorithm/rotate
  */
 template<class T, class BI>
-void rotate(std::list<T>& list, BI begin, BI new_begin, BI end);
+void rotate(std::list<T>& list, BI begin, BI new_begin, BI end)
+{
+    list.splice(list.begin(), list, std::next(begin, std::distance(begin, new_begin)), end);
```

Comment:
All needed iterators are passed to this algorithm!
Use them.

Reply:
updated